nrf52840-hal/src/ieee802154.rs

```diff
@@ -455,9 +477,14 @@ impl<'c> Radio<'c> {
                     .events_ccabusy()
                     .bit_is_set()
                 {
-                    // channel is busy; try another CCA
-                    self.radio.events_ccabusy.reset();
-                    continue 'cca;
+                    if num_retries_left == 0 {
+                        break 'cca;
```

Comment:
this should return an error and do proper cleanup (disable shorts) but that would require changing the signature of `send` to return a `Result`. As implemented this method may silently fail to send a packet.

I would prefer to *not* modify `send` in this PR. `send` is explicitly document as not being spec compliant (due to the lack of backoff delays). Adding this makes it a bit more spec compliant but still not quite spec compliant. I would prefer to keep it as a non-compliant utility method. People that want a spec-compliant `send` should build one on top of the `try_send` method (which is probably still hard to do due to lack of (T)IFS support).